scalameta/contrib/src/main/scala/scala/meta/contrib/ScaladocParser.scala

```diff
@@ -0,0 +1,120 @@
+package scala.meta.contrib
+
+import DocToken._
+import scala.meta.Syntax
+import scala.meta.tokens.Token.Comment
+
+object ScaladocParser {
+
+  /**
+    * Parses a scaladoc [[Comment]].
+    */
+  def parseScaladoc(comment: Comment): Seq[DocToken] = {
+
+    val trimmedComment: String = comment.show[Syntax].trim
+
+    require(
+      requirement = trimmedComment.startsWith("/*") && trimmedComment.endsWith("*/"),
+      message = "Input comment is not a Scaladoc"
+    )
+
+    val scaladocSymbols = Seq('/', '*', ' ')
+
+    val lineSeparedDocString: Seq[String] =
+      trimmedComment
+        .split(Array('\n', '\r')) // Splits the Scaladoc in lines
```

Comment:
@xavier-fernandez 

Here's one example way to approach it

```
@ val codeblock = P("{{{" ~ ((AnyChar~ !"}}}").rep ~ AnyChar).! ~ "}}}")
codeblock: Parser[String] = codeblock
@ val doc = """{{{
      def codeBlockBody(foo: Bar)
  }}}
  Bla bla bla
  """"
doc: String = """
{{{
    def codeBlockBody(foo: Bar)
}}}
Bla bla bla
"
"""
@ codeblock.parse(doc)
res42: Parsed[String, Char, String] = Success(
  """

    def codeBlockBody(foo: Bar)

  """,
  39
)
```

The `codeblock` parser starts at the `{{{` and ends at the `}}}` and consumes only the input inside the blocks.

I takes a while get the hang of fastparse but once it clicks it feels very natural. Fastparse is awesome, I highly recommend the excellent docs http://www.lihaoyi.com/fastparse/